### ISSUE TITLE
[enhancement]: Use `@tsconfig/node10` for `tsconfig.json`

### ISSUE THREAD

**Issue by JamieMagee** [microsoft/azure-pipelines-tasks#18118]
### Describe your feature request here

Currently all tasks have a minimal `tsconfig.json` which targets `ES6` (AKA ES2015) that looks something like this:

```json
{
  "compilerOptions": {
      "target": "ES6",
      "module": "commonjs"
  }
}
```

There are some variations here and there for different tasks, and `AzureIotEdgeV2` targets `ES2016`[^4] (AKA `ES7`) for some unknown reason.

`ES6` was a sensible choice when Azure Pipelines was limited to Node 6, which limited tasks to `ES6`[^1], but now all of the tasks in this repository target Node 10, and the majority target Node 16 as well. By continuing to target `ES6` the TypeScript compiler generates code that uses a more restricted feature set than if you were to set an appropriate target for the version of Node in use. This means more verbose, worse performing code.

For Node 10 the appropriate target is `ES2018`[^2] and for Node 16 the appropriate target is `ES2021`[^3].

Thankfully, the TypeScript community has come together and created [tsconfig/bases][1], which contain recommended `tsconfig.json` settings for each Node version. For example [`@tsconfig/node10`][2] maps to the following configuration

```json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Node 10",

  "compilerOptions": {
    "lib": ["es2018"],
    "module": "commonjs",
    "target": "es2018",

    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "node"
  }
}
```

There is also a configuration for [`@tsconfig/node16`][3], and a configuration [`@tsconfig/strictest`][4], which enables all of the strict compiler configuration.

In summary, my suggestion is to migrate away from custom configuration to `@tsconfig/node10` for improved performance and ease of maintenance.

[1]: https://github.com/tsconfig/bases
[2]: https://www.npmjs.com/package/@tsconfig/node10
[3]: https://www.npmjs.com/package/@tsconfig/node16
[4]: https://www.npmjs.com/package/@tsconfig/strictest

[^1]: https://node.green/
[^2]: https://github.com/tsconfig/bases/blob/7db25a41bc5a9c0f66d91f6f3aa28438afcb2f18/bases/node10.json#L8
[^3]: https://github.com/tsconfig/bases/blob/7db25a41bc5a9c0f66d91f6f3aa28438afcb2f18/bases/node16.json#L8
[^4]: https://github.com/microsoft/azure-pipelines-tasks/blob/65a9f6e8d367eff31d6e50ad871e0fd62aa6c541/Tasks/AzureIoTEdgeV2/tsconfig.json#L3

**Comment by JamieMagee** [comment]
This doesn't appear to be complete.
